app/views.py

```diff
@@ -34,9 +34,10 @@ class MyModelView(ModelView):
 """
 
 
-class UploadsModelView(CompactCRUDMixin, ModelView):
-    datamodel = SQLAInterface(Uploads)
-    base_permissions = ['can_add', 'can_list', 'can_show']
+class UploadModelView(CompactCRUDMixin, ModelView):
+    datamodel = SQLAInterface(Upload)
+    base_permissions = ['can_add', 'can_list', 'can_show', 'can_download']
```

Comment:
@Adman adding `can_download` should fix the "Access Denied" issues you were seeing. Sorry about that.